shell/platform/darwin/macos/framework/Source/FlutterEngineTestUtils.mm

```diff
@@ -4,11 +4,44 @@
 // found in the LICENSE file.
 
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngineTestUtils.h"
+
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterDartProject_Internal.h"
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
 
 #include "flutter/testing/testing.h"
 
+// Fake pasteboard implementation to allow tests to work in environments without a real
+// pasteboard.
+@interface FakePasteboard : FlutterPasteboard
+@end
+
+@implementation FakePasteboard {
+  NSString* _result;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self != nil) {
+    _result = nil;
+  }
+  return self;
+}
+
+- (NSInteger)clearContents {
+  return 1;
```

Comment:
Maybe comment that this is ignored by our implementation? The use of 1 makes it seem like the value might matter.

Reply:
Ah good point! Added a "proper" fake implementation that returns 0 if _result was already nil and 1 otherwise. Seems unlikely we'll ever care about the result of this, but it's minimal effort to implement so may as well, just in case we use it one day.